src/EnergyPlus/WaterToWaterHeatPumpEIR.cc

```diff
@@ -0,0 +1,517 @@
+                                                                                      cCurrentModuleObject,
+                                                                                      thisWWHP.name,
+                                                                                      DataLoopNode::NodeType_Water,
+                                                                                      DataLoopNode::NodeConnectionType_Outlet,
+                                                                                      flowPath2,
+                                                                                      DataLoopNode::ObjectIsNotParent);
+                BranchNodeConnections::TestCompSet(
+                    cCurrentModuleObject, thisWWHP.name, loadSideInletNodeName, loadSideOutletNodeName, "Hot Water Nodes");
+                BranchNodeConnections::TestCompSet(
+                    cCurrentModuleObject, thisWWHP.name, sourceSideInletNodeName, sourceSideOutletNodeName, "Condenser Water Nodes");
+
+                eir_wwhp.push_back(thisWWHP);
+            }
+        }
+    }
+
+
+    void EIRWaterToWaterHeatPump::processInputForEIRWWHPCooling()
+    {
+        using namespace DataIPShortCuts;
+
+        bool errorsFound = false;
+
+        cCurrentModuleObject = "HeatPump:WaterToWater:EIR:Cooling";
+        int numWWHP = inputProcessor->getNumObjectsFound(cCurrentModuleObject);
+        if (numWWHP > 0) {
+            auto const instances = inputProcessor->epJSON.find(cCurrentModuleObject);
+            if (instances == inputProcessor->epJSON.end()) {
+                errorsFound = true;
+            }
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = instance.key();
+
+                EIRWaterToWaterHeatPump thisWWHP;
+                thisWWHP.plantTypeOfNum = DataPlant::TypeOf_HeatPumpEIRCooling;
+                thisWWHP.name = UtilityRoutines::MakeUPPERCase(thisObjectName);
+                std::string loadSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("load_side_inlet_node_name"));
+                std::string loadSideOutletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("load_side_outlet_node_name"));
+                std::string sourceSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("source_side_inlet_node_name"));
+                std::string sourceSideOutletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("source_side_outlet_node_name"));
+                thisWWHP.loadSideDesignVolFlowRate = fields.at("load_side_reference_flow_rate");
+                thisWWHP.sourceSideDesignVolFlowRate = fields.at("source_side_reference_flow_rate");
+                thisWWHP.referenceCapacity = fields.at("reference_capacity");
```

Comment:
You realize these will crash when Autosize is entered in DesignVolFlowRate and Capacity fields.